src/EventListener/AssetsInstallCommandListener.php

```diff
@@ -0,0 +1,254 @@
+
+    protected static $defaultName = 'assets:install';
+
+    private $filesystem;
+    private $projectDir;
+
+    public function __construct(Filesystem $filesystem, ?string $projectDir = null)
+    {
+        if (null === $projectDir) {
+            @trigger_error(sprintf('Not passing the project directory to the constructor of %s is deprecated since Symfony 4.3 and will not be supported in 5.0.', __CLASS__), E_USER_DEPRECATED);
+        }
+
+        $this->filesystem = $filesystem;
+        $this->projectDir = $projectDir;
+    }
+
+    public function copySonataCoreAssets(ConsoleTerminateEvent $event)
+    {
+        $command = $event->getCommand();
+        $application = $command->getApplication();
+
+        try {
+            $coreBundle = $application->getKernel()->getBundle('SonataCoreBundle');
+        } catch (\Exception $e) {
+            $coreBundle = null;
+        }
+
+        if ('assets:install' !== $command->getName() || null !== $coreBundle) {
+            return;
+        }
+
+        $output = $event->getOutput();
+
+        $this->execute($event->getInput(), $event->getOutput(), $application);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output, Application $application): int
+    {
+        /**
+         * @var KernelInterface
+         */
+        $kernel = $application->getKernel();
+
+        $targetArg = rtrim($input->getArgument('target') ?? '', '/');
+
+        if (!$targetArg) {
+            $targetArg = $this->getPublicDirectory($kernel->getContainer());
+        }
+
+        if (!is_dir($targetArg)) {
+            $targetArg = $kernel->getProjectDir().'/'.$targetArg;
+
+            if (!is_dir($targetArg)) {
+                throw new InvalidArgumentException(sprintf('The target directory "%s" does not exist.', $input->getArgument('target')));
+            }
+        }
+
+        $bundlesDir = $targetArg.'/bundles/';
+
+        $io = new SymfonyStyle($input, $output);
+        $io->newLine();
+
+        if ($input->getOption('relative')) {
+            $expectedMethod = self::METHOD_RELATIVE_SYMLINK;
+            $io->text('Trying to install deprecated SonataCoreBundle assets from SonataAdminBundle as <info>relative symbolic links</info>.');
+        } elseif ($input->getOption('symlink')) {
+            $expectedMethod = self::METHOD_ABSOLUTE_SYMLINK;
+            $io->text('Trying to install deprecated SonataCoreBundle assets from SonataAdminBundle as <info>absolute symbolic links</info>.');
+        } else {
+            $expectedMethod = self::METHOD_COPY;
+            $io->text('Installing deprecated SonataCoreBundle assets from SonataAdminBundle as <info>hard copies</info>.');
+        }
+
+        $io->newLine();
+
+        $copyUsed = false;
+        $exitCode = 0;
+        $validAssetDirs = [];
+
+        $bundle = $kernel->getBundle('SonataAdminBundle');
+        $originDir = $bundle->getPath().'/Resources/sonataCoreAssets';
```

Comment:
Is there a reason why this is called `sonataCoreAssets` and not `SonataCoreAssets` or just `SonataCoreBundle`?

Reply:
I just name it like this. SonataCoreBundle will be fonfusing. I will change it to SonataCoreBundleAssets when i finish clearing my computer from dust.

Reply:
I just name it like this. SonataCoreBundle will be fonfusing. I will change it to SonataCoreBundleAssets when i finish clearing my computer from dust.